sycl/test-e2e/ESIMD/unified_memory_api/Inputs/block_store.hpp

```diff
@@ -219,6 +218,164 @@ bool testACC(queue Q, uint32_t Groups, uint32_t Threads,
   return Passed;
 }
 
+template <typename T, uint16_t N, bool UseMask, bool CheckProperties,
+          typename StorePropertiesT>
+bool testSLM(queue Q, uint32_t Groups, StorePropertiesT StoreProperties) {
+  using shared_allocator = sycl::usm_allocator<T, sycl::usm::alloc::shared, 16>;
+  using shared_vector = std::vector<T, shared_allocator>;
+  constexpr uint16_t GroupSize = 8;
+
+  uint32_t Size = Groups * GroupSize * N;
+
+  std::cout << "SLM case: T=" << esimd_test::type_name<T>() << ",N=" << N
+            << ",UseMask=" << UseMask << ",UseProperties=" << CheckProperties
+            << std::endl;
+
+  sycl::range<1> GlobalRange{Groups};
+  sycl::range<1> LocalRange{GroupSize};
+  sycl::nd_range<1> Range{GlobalRange * LocalRange, LocalRange};
+
+  constexpr size_t Alignment = getAlignment<T, N, UseMask>(StoreProperties);
+
+  shared_vector Out(GroupSize, shared_allocator{Q});
+  T Out_val = esimd_test::getRandomValue<T>();
+  for (int i = 0; i < Size; i++)
+    Out[i] = Out_val;
+
+  try {
+    Q.submit([&](handler &CGH) {
+       auto OutPtr = Out.data();
+
+       CGH.parallel_for(Range, [=](sycl::nd_item<1> ndi) SYCL_ESIMD_KERNEL {
+         constexpr uint32_t SLMSize = (GroupSize * N) * sizeof(T);
+         slm_init<SLMSize>();
+
+         uint16_t GlobalID = ndi.get_global_id(0);
+         uint16_t LocalID = ndi.get_local_id(0);
+         uint32_t LocalElemOffset = LocalID * N * sizeof(T);
+         uint32_t ElemOff = GlobalID * N;
+         simd<T, N> Vals(ElemOff, 1);
+
+         if constexpr (UseMask) {
+           simd_mask<1> Mask = (GlobalID + 1) & 0x1;
+           simd<T, N> PassThru = Out_val;
+           if constexpr (CheckProperties)
+             slm_block_store(LocalElemOffset, Vals, Mask, StorePropertiesT{});
+           else
+             slm_block_store(LocalElemOffset, Vals, Mask);
+           Vals = slm_block_load<T, N>(LocalElemOffset, Mask, PassThru);
```

Comment:
The test is targeted to block_store.
Loading with the same mask and PassThru does not let you to check if 'mask' in block_store is really honored. If it is ignored, the test will not catch it.

Better logic would be:
- 0-th thread initializes SLM with some value (or values).
- barrier
- step3: slm_block_store writes to SLM with a mask
- block_load _without_ a mask is called to read whatever is stored in SLM at that moment.
- verification: It also checks that 'mask' on step3 is honored.